dropwizard-jetty/src/main/java/io/dropwizard/jetty/ZipExceptionHandlingInputStream.java

```diff
@@ -0,0 +1,97 @@
+package io.dropwizard.jetty;
+
+import java.io.EOFException;
+import java.io.FilterInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.zip.ZipException;
+
+import org.eclipse.jetty.http.BadMessageException;
+
+/**
+ * This InputStream is used to decorate a GZIPInputStream or InflaterInputStream, intercept decompression
+ * exceptions, and rethrow them as BadMessageExceptions
+ */
+class ZipExceptionHandlingInputStream extends FilterInputStream {
+
+    private final String format;
+
+    ZipExceptionHandlingInputStream(InputStream in, String format) {
+        super(in);
+        this.format = format;
+    }
+
+    @Override
+    public int read(byte[] b, int off, int len) throws IOException {
+        try {
+            return super.read(b, off, len);
+        } catch (ZipException e) {
```

Comment:
I think it makes sense  to add a generic method for handling an `IOException` instead of repeat the logic in every method. Something like:
```java
static BadMessageException handleException(String format, IOException e) throws IOException {
        if (e instanceof ZipException) {
            return buildBadDataException(format, e);
        } else if (e instanceof EOFException) {
            return buildPrematureEofException(format, e);
        } else {
            throw e;
        }
}
```